### ISSUE TITLE
Migration to 1.4 from 1.3.x doesn't preserve the current root correctly.

### ISSUE THREAD

**Issue by kensipe** [mesosphere/marathon#4791]
Steps to duplicate:
1. fresh dcos cluster
2. install 1.3.6
3. launch task (example below)
4. after it is running, destroy the app
5. uninstall marathon 1.3.6
6. install 1.4.0-snap30
(The 3 tasks that were removed "destroyed" in marathon 1.3.6 will start running again)

```
{
      "id": "test3",
      "instances":  3,
      "cmd": "for (( ; ; )); do sleep 100000000; done",
      "cpus": 0.01,
      "mem": 1,
      "disk": 0
    }
```

**Comment by kensipe** [comment]
Additional note:  The tasks will be started up and will be running when viewed from dcos or mesos... however they do NOT show up in the list of apps in `v2/apps` as they are under a different framework_id.

**Comment by jdef** [comment]
upon uninstallation of marathon 1.3.6, are the tasks actually destroyed? or do they keep running in the background? I would expect uninstall to effectively terminate (on mesos) any existing tasks associated with 1.3.6. if that's not happening, and these tasks just keep running, then mesos is probably waiting for some scheduler failover period before killing the executors running the task

**Comment by kensipe** [comment]
I'll run through the scenario again... the tasks were KILLED from a mesos standpoint  from what I remember.

**Comment by jasongilanfarr** [comment]
I see the issue, prepping a fix.

**Comment by jdef** [comment]
currently suspect a bug in migration:
(1) migration report indicates that only root+apps have been migrated (counts for other objects are all zero)
(2) mesos and marathon logs indicate that `marathon-user` (MoM) tasks are KILLED and removed prior to framework shutdown

@kensipe has mentioned that some tasks show as "orphaned" via mesos `/state`. Not sure what's happening there. Possibly a bug in reconciliation, but needs further investigation.

**Comment by jdef** [comment]
xref #4581

**Comment by unterstein** [comment]
Should be fixed with fix for #4824. Will test this now
